### PR TITLE
Modularize: Minor PR

### PR DESCRIPTION
**Pull request by vincentarelbundock**
A minor PR with some more modularization/simplification:

* Removed duplicated code
* Removed unused arguments
* `type_*()` data preparation function are now stored in a dictionary with switch-style selection.

### CODE REVIEW

File: R/by_aesthetics.R [hunk 1]
```diff
@@ -247,3 +247,47 @@ by_lwd = function(ngrps, type, lwd=NULL) {
...
+  by_continuous,
+  by_ordered,
+  col,
+  fill,
+  ngrps,
+  palette,
+  ribbon.alpha,
+  type) {
+
+  if (is.null(bg) && !is.null(fill)) bg = fill
```
Comment (grantmcdermott) [inline R/by_aesthetics.R:266]: Should this go higher up in the main tinyplot function? That way, we can also drop the extra fill argument for this `by_bg` function (as well as the input on l. 665 below).

**Review by grantmcdermott — CHANGES_REQUESTED** [review]
Very nice, thanks. One minor comment and then happy to merge.

**Review by grantmcdermott — APPROVED** [review]

**Comment by grantmcdermott** [comment]
Not so minor in the end... thanks again for the continued efficiency gains!

**Comment by vincentarelbundock** [comment]
no it really was minor. I just noticed that your suggestion could help us simplify something else too.

This was a simple PR, but I'll take the 35 lines of code removed :)

**Pull request merged**